src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -144,7 +146,7 @@ public FormValidation doCheckUrl(@AncestorInPath Item project,
                 return FormValidation.ok();
 
             // get git executable on master
-            final EnvVars environment = new EnvVars(System.getenv()); // GitUtils.getPollEnvironment(project, null, launcher, TaskListener.NULL, false);
+            final EnvVars environment = project.getEnvironment(Jenkins.getActiveInstance(), TaskListener.NULL);
```

Comment:
I think my concern is more around some of the env changes that can occur in the project which are intended for the remote system and might redefine the git tool location via something like an EnvironmentContributor (ugh).  

Maybe there's an angle I'm not seeing that prevents that possibility, and I'll admit it's slightly obscure.  I'm trying to figure out what we're gaining in this place by using the project env (which ties back to the JDK env) vs. the Process environment with System.getEnv.   Maybe there's a nuance there I'm missing?